src/librustc_mir/interpret/eval_context.rs

```diff
@@ -291,22 +291,37 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         ty.is_freeze(*self.tcx, self.param_env, DUMMY_SP)
     }
 
-    pub(super) fn subst_and_normalize_erasing_regions<T: TypeFoldable<'tcx>>(
+    /// Call this whenever you have a value that `needs_subst`. Not guaranteed to actually
```

Comment:
This advice seems wrong. It should be called whenever (and only when) you took something from the MIR of the current frame.

Maybe something like
```
    /// Call this on things you got out of the MIR (so it is as generic as the current
    /// stack rameframe), to bring it into the proper environment for this interpreter.
    ///
    /// This is not guarantee to actually monomorphize [...]
```